src/NuGet.Core/NuGet.Packaging/Signing/SigningUtility.cs

```diff
@@ -233,6 +248,95 @@ public static bool IsValidForPurposeFast(X509Certificate2 certificate, string ek
 
             policy.VerificationTime = verificationTime;
         }
+
+        public static bool IsCertificateValidityPeriodInTheFuture(X509Certificate2 certificate)
+        {
+            DateTimeOffset signerCertBegin = DateTime.SpecifyKind(certificate.NotBefore, DateTimeKind.Local);
+
+            return signerCertBegin > DateTimeOffset.Now;
+        }
+
+        internal static bool BuildCertificateChain(X509Chain chain, X509Certificate2 certificate, out X509ChainStatus[] status)
+        {
+            if (certificate == null)
+            {
+                throw new ArgumentNullException(nameof(certificate));
+            }
+
+            var buildSuccess = chain.Build(certificate);
+            status = new X509ChainStatus[chain.ChainStatus.Length];
+            chain.ChainStatus.CopyTo(status, 0);
+
+            // Check if time is not in the future
+            return buildSuccess && !IsCertificateValidityPeriodInTheFuture(certificate);
+        }
+
+        internal static bool IsTimestampValid(byte[] data, bool failIfInvalid, List<SignatureLog> issues, SignerInfo timestampSignerInfo, Rfc3161TimestampTokenInfo tstInfo, SigningSpecifications spec)
```

Comment:
This method is not referenced anywhere.

Reply:
Referenced [here](https://github.com/NuGet/NuGet.Client/pull/1868/files#diff-ce902b6f1ff63e49285926ee427d5bf2R215)